netket/optimizer/qgt/qgt_jacobian.py

```diff
@@ -81,11 +91,11 @@ def __matmul__(self, vec: Union[PyTree, jnp.ndarray]) -> Union[PyTree, jnp.ndarr
     def _unscaled_matmul(self, vec: jnp.ndarray) -> jnp.ndarray:
         return (
             sum_inplace(((self.O @ vec).T.conj() @ self.O).T.conj())
-            + self.sr.diag_shift * vec
+            + self.diag_shift * vec
         )
 
     @jax.jit
-    def solve(self, y: PyTree, x0: Optional[PyTree] = None) -> PyTree:
+    def _solve(self, solve_fun, y: PyTree, *, x0: Optional[PyTree] = None) -> PyTree:
```

Comment:
should also pass `**kwargs` and add them to the calling sequence of `solve_fun` at the end

Reply:
I'm not so sure about this.
The problem with kwargs is that we don't know if they are static or not.
So we cannot really propagate them inside the jax.jit

Reply:
Fair enough. I guess this fits with the overall idea that the solver passed to `Vmc` is `partial`ed with every kwarg the user wants to pass... 
Will this be a good override for `_solve` in the base class though, seeing that one passes kwargs?

Reply:
this in principle is also a good idea.
But how do we wrap the kwargs? as static parameters or as runtime parameters?
For example, maxiter is a static argument.
The tolerance should not be a static argument

Reply:
Forgive my ignorance of python syntax, what does the `*` do here? I just wonder if `x0` gets passed on correctly

Reply:
* means that all arguments after that must be passed as keyword and cannot be passed positionally